src/mongocxx/options/aggregate.cpp

```diff
@@ -33,6 +36,8 @@ aggregate& aggregate::batch_size(std::int32_t batch_size) {
 }
 
 aggregate& aggregate::max_time(std::chrono::milliseconds max_time) {
+    if (!is_int32_duration<std::chrono::milliseconds>(max_time))
+        throw logic_error{error_code::k_invalid_parameter};
```

Comment:
You can provide a message with logic_error, and it might be good to do so. If you don't want to repeat the text, just make a constant string in the header where you define is_int32_duration:

const char kDurationOverflow[] = "A chrono duration was provided that exceeds the implementation limits";